src/System.Data.SqlClient/src/System/Data/SqlClient/SqlCommand.cs

```diff
@@ -290,7 +293,8 @@ internal SqlStatistics Statistics
             {
                 if (null != _activeConnection)
                 {
-                    if (_activeConnection.StatisticsEnabled)
+                    if (_activeConnection.StatisticsEnabled || 
+                        _diagnosticListener.IsEnabled(SqlClientDiagnosticListenerExtensions.SqlAfterExecuteCommand))
```

Comment:
A listener is _enabled_ when an observer is registered and self-reports as listening to the diagnosticListener. So in practice it will be disabled by default and then enabled once an interested observer shows up.